src/test/lib/operators/operator_scan_predicate_test.cpp

```diff
@@ -83,6 +84,28 @@ TEST_F(OperatorScanPredicateTest, FromExpressionColumnRight) {
   EXPECT_EQ(operator_predicate_a.value, AllParameterVariant{5});
 }
 
+TEST_F(OperatorScanPredicateTest, OutputToStream) {
+  auto test_cases = std::vector<std::tuple<std::optional<std::vector<OperatorScanPredicate>>, std::string>>();
+  test_cases.push_back(make_tuple(OperatorScanPredicate::from_expression(*between_inclusive_(5, a, b), *node),
```

Comment:
Can we set the `operatorscanpredicate::from_expression()` within the loop? `operatorscanpredicate::from_expression(` and `, *node)` is alway the same.

You can also directly instantiate the vector `const auto test_cases = std::vector<std::tuple>>{ {..., ..., ...} }`. If we really only need two items, a `std::pair` would be a better fit than the tuple.

Reply:
I've tried to do this, but I got stuck setting up the types of the vector. Maybe you can help me? I don't know how to specify the types of the `expression_functional::detail::ternary<PredicateCondition, AbstractPredicateExpression` as the PredicateConditions vary. 

```
  const auto test_cases = std::vector<std::pair<expression_functional::detail::ternary<PredicateCondition, AbstractPredicateExpression>, std::string>>({
     {*between_inclusive_(5, a, b),
      "Column #0 <=5\nColumn #1 >=5\n"},
     {*greater_than_(a, 5), "Column #0 >5\n"},
     {*less_than_(a, 5), "Column #0 <5\n"},
     {*greater_than_(a, b), "Column #0 >Column #1\n"}
  });
```

Reply:
`std::shared_ptr<AbstractExpression>`/`std::shared_ptr<AbstractPredicateExpression>`?

Reply:
`std::shared_ptr<AbstractExpression>`/`std::shared_ptr<AbstractPredicateExpression>`?

Reply:
Worked 👍 thanks!